flang/include/flang/Common/format.h

```diff
@@ -214,15 +215,13 @@ template <typename CHAR> void FormatValidator<CHAR>::NextToken() {
   case '7':
   case '8':
   case '9': {
-    int64_t lastValue;
     const CHAR *lastCursor;
     integerValue_ = 0;
     bool overflow{false};
     do {
-      lastValue = integerValue_;
       lastCursor = cursor_;
       integerValue_ = 10 * integerValue_ + c - '0';
-      if (lastValue > integerValue_) {
+      if (integerValue_ > std::numeric_limits<int>::max()) {
```

Comment:
How will this new condition ever be true?

The idea is that an _n+1_ digit unsigned number should always be larger than an _n_ digit unsigned number. With bounded computer arithmetic that fails if the number overflows.

Reply:
`integerValue_` is std::int64_t

Reply:
`integerValue_` hasn't overflowed its 64-bit range, and can be compared to the max 32-bit value safely.

Reply:
Why restrict values to 32 bits? Don't we typically accommodate 64 values by "default"?

Reply:
The checker in format-implementation.h limited things to 32 bits already.  That seems sufficient to me for repeat counts, field widths, &c. in FORMAT strings.

Reply:
So the two cases are not then strictly comparable. Maybe the best choice is to leave the compile time format.h code as is and go with new code in the runtime.